driver-core/src/test/java/com/datastax/driver/core/SchemaChangesTest.java

```diff
@@ -39,24 +61,35 @@
 
     Session session;
 
+    SchemaChangeTestListener listener1;
+    SchemaChangeTestListener listener2;
+
+    List<SchemaChangeTestListener> listeners;
+
     @BeforeClass(groups = "short")
-    public void setup() {
+    public void setup() throws InterruptedException {
         ccm = CCMBridge.builder("schemaChangesTest").withNodes(1).build();
 
-        cluster = Cluster.builder().addContactPoint(CCMBridge.ipOfNode(1)).build();
+        cluster1 = Cluster.builder().addContactPoint(CCMBridge.ipOfNode(1)).build();
         cluster2 = Cluster.builder().addContactPoint(CCMBridge.ipOfNode(1)).build();
 
-        metadatas = Lists.newArrayList(cluster.getMetadata(), cluster2.getMetadata());
+        metadatas = Lists.newArrayList(cluster1.getMetadata(), cluster2.getMetadata());
+
+        session = cluster1.connect();
+
+        cluster1.register(listener1 = spy(new SchemaChangeTestListener()));
+        cluster1.register(listener2 = spy(new SchemaChangeTestListener()));
```

Comment:
Combining a spy and a custom implementation is a bit convoluted.
I understand that the custom code provides a way to wait for the async event, but we could use a pure mock [timeout()](http://docs.mockito.googlecode.com/hg/org/mockito/verification/VerificationWithTimeout.html) for the same purpose.